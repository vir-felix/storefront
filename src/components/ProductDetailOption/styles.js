export default (theme) => ({
  variantOptionContainer: {
    display: "flex",
    width: `${theme.spacing.unit * 3.125}%`,
    maxWidth: `${theme.spacing.unit * 3.125}%`,
    padding: theme.spacing.unit * 1.25
  },
  optionButton: {
    "flex": "1 1 auto",
    "fontWeight": 600,
    "padding": theme.spacing.unit * 1.25,
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.borderColor,
    "&:hover": {
      backgroundColor: theme.palette.reaction.activeElementBackground
    }
  },
  optionText: {
    fontWeight: 500,
    fontSize: "14px"
  },
  isSelected: {
    backgroundColor: theme.palette.reaction.activeElementBackground
  }
});
