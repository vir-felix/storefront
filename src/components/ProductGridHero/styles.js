export default (theme) => ({
  heroImg: {
    width: "100%",
    height: "325px",
    objectFit: "cover"
  },
  productGridContainer: {
    maxWidth: theme.grid.productGridMaxWidth,
    marginLeft: "auto",
    marginRight: "auto"
  }
});
