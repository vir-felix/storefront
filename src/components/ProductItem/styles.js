export const styles = (theme) => ({
  "@keyframes spin": {
    from: { transform: "rotate(0deg)" },
    to: { transform: "rotate(-360deg)" }
  },
  "root": {},
  "productInfo": {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    paddingTop: 10
  },
  "productMedia": {
    backgroundColor: theme.palette.primary.contrastText,
    position: "relative"
  },
  "img": {
    height: "auto",
    width: "100%"
  },
  "imgLoading": {
    alignItems: "center",
    backgroundColor: "transparent",
    display: "flex",
    fontSize: 48,
    height: 100,
    justifyContent: "center",
    left: "calc(50% - 50px)",
    position: "absolute",
    top: "calc(50% - 50px)",
    width: 100
  },
  "loadingIcon": {
    color: theme.palette.primary.light,
    fontSize: "inherit",
    animationName: "spin",
    animationDuration: theme.transitions.duration.standard * 2,
    animationTimingFunction: theme.transitions.easing.sharp,
    animationIterationCount: "infinite",
    animationFillMode: "both"
  },
  "link": {
    ...theme.typography.body2,
    "textAlign": "left",
    "&:hover": {
      color: theme.palette.secondary.main
    }
  }
});
