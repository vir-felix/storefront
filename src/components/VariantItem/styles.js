export default (theme) => ({
  variantButton: {
    "display": "flex",
    "justifyContent": "space-between",
    "textTransform": "none",
    "width": "100%",
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.contrastText,
    "paddingTop": theme.spacing.unit * 1.25,
    "paddingBottom": theme.spacing.unit * 1.25,
    "paddingLeft": theme.spacing.unit * 2.5,
    "paddingRight": theme.spacing.unit * 2.5,
    "&:hover": {
      border: "1px solid",
      borderColor: theme.palette.reaction.activeElementBorderColor,
      backgroundColor: theme.palette.primary.contrastText
    }
  },
  activeVariant: {
    border: "1px solid",
    borderColor: theme.palette.reaction.activeElementBorderColor
  }
});
