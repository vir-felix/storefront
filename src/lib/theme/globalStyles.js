export default
<style global jsx>
  {`
    a {
      color: inherit;
      text-decoration: none;
    }
  `}
</style>;
