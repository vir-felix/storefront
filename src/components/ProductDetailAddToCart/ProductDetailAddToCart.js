import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import ButtonBase from "material-ui/ButtonBase";
import Grid from "material-ui/Grid";
import InputAdornment from "material-ui/Input/InputAdornment";
import TextField from "material-ui/TextField";
import Typography from "material-ui/Typography";
import Minus from "mdi-material-ui/Minus";
import Plus from "mdi-material-ui/Plus";
import { inject, observer } from "mobx-react";
import Divider from "components/Divider";

const styles = (theme) => ({
  addToCartButton: {
    "padding": theme.spacing.unit,
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.light,
    "minWidth": "66%",
    "&:hover": {
      borderColor: theme.palette.reaction.activeElementBorderColor
    },
    "&:focus": {
      outline: "auto 5px -webkit-focus-ring-color"
    }
  },
  addToCartText: {
    color: theme.palette.primary.contrastText,
    fontWeight: 600
  },
  incrementButton: {
    backgroundColor: theme.palette.reaction.activeElementBackground,
    padding: 6
  },
  quantityContainer: {
    padding: 0,
    border: `1px solid ${theme.palette.reaction.borderColor}`,
    backgroundColor: theme.palette.common.white,
    borderRadius: theme.palette.reaction.buttonBorderRadius
  },
  quantityGrid: {
    marginBottom: theme.spacing.unit * 3
  },
  quantityInput: {
    "width": "40px",
    "textAlign": "center",
    "&:focus": {
      borderColor: "#80bdff",
      boxShadow: "0 0 0 0.2rem rgba(0,123,255,.25)"
    },
    "borderLeft": `1px solid ${theme.palette.reaction.borderColor}`,
    "borderRight": `1px solid ${theme.palette.reaction.borderColor}`
  },
  quantityTypography: {
    marginBottom: theme.spacing.unit * 2
  }
});


@withStyles(styles, { withTheme: true })
@inject("pdpStore")
@observer
export default class ProductDetailAddToCart extends Component {
  static propTypes = {
    classes: PropTypes.object,
    onClick: PropTypes.func.isRequired,
    option: PropTypes.object,
    selectedOption: PropTypes.string
  }

  static propTypes = {
    classes: PropTypes.object,
    pdpStore: PropTypes.object
  };

  static defaultProps = {
    classes: {},
    pdpStore: {}
  };

  state = {
    addToCartQuantity: 1
  };

  handleOnClick = () => {
    // This function currently does nothing. When our GraphQL endpoints are available, we'll use them to add the items to the cart.
    // To test that this is working, uncomment the following lines and check to see that the data is correct

    // const { pdpStore } = this.props;
    // const { addToCartQuantity } = this.state;
    // console.log("Selected Option", pdpStore.selectedOption);
    // console.log("Quantity", addToCartQuantity);

    // Reset cart quantity to 1 after items are added to cart
    this.setState({ addToCartQuantity: 1 });
  }

  handleQuantityInputChange = (event) => {
    const { value } = event.target;

    this.setState({ addToCartQuantity: value });
  }

  handleIncrementButton = () => {
    const value = this.state.addToCartQuantity + 1;

    this.setState({ addToCartQuantity: value });
  }

  handleDecrementButton = () => {
    const value = this.state.addToCartQuantity - 1;

    if (value >= 0) {
      this.setState({ addToCartQuantity: value });
    }
  }

  render() {
    const {
      classes: {
        addToCartButton,
        addToCartText,
        incrementButton,
        quantityContainer,
        quantityGrid,
        quantityInput,
        quantityTypography
      }
    } = this.props;

    const { addToCartQuantity } = this.state;

    return (
      <Grid container>
        <Grid item xs={12} className={quantityGrid}>
          <Divider />
          <Typography component="span" className={quantityTypography}>Quantity</Typography>
          <TextField
            id="addToCartQuantityInput"
            value={addToCartQuantity}
            onChange={this.handleQuantityInputChange}
            InputProps={{
              startAdornment: (
                <InputAdornment>
                  <ButtonBase
                    color="default"
                    variant="outlined"
                    onClick={this.handleDecrementButton}
                    className={incrementButton}
                  >
                    <Minus />
                  </ButtonBase>
                </InputAdornment>
              ),
              endAdornment: (
                <InputAdornment>
                  <ButtonBase
                    variant="outlined"
                    color="default"
                    onClick={this.handleIncrementButton}
                    className={incrementButton}
                  >
                    <Plus />
                  </ButtonBase>
                </InputAdornment>
              ),
              disableUnderline: true,
              classes: {
                root: quantityContainer,
                input: quantityInput
              }
            }}
          />
        </Grid>
        <Grid item xs={12}>
          <ButtonBase
            onClick={this.handleOnClick}
            className={addToCartButton}
          >
            <Typography className={addToCartText} component="span" variant="body1">
              Add to cart
            </Typography>
          </ButtonBase>
        </Grid>
      </Grid>
    );
  }
}
