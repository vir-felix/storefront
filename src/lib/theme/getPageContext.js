import { SheetsRegistry } from "jss";
import { createGenerateClassName } from "@material-ui/core/styles";
import theme from "./reactionTheme";

/**
 * @summary Creates and returns a page context object
 * @returns {Object} Page context object
 */
function createPageContext() {
  return {
    theme,
    // This is needed in order to deduplicate the injection of CSS in the page.
    sheetsManager: new Map(),
    // This is needed in order to inject the critical CSS.
    sheetsRegistry: new SheetsRegistry(),
    // The standard class name generator.
    generateClassName: createGenerateClassName()
  };
}

/**
 * Either creates a new or returns an existing page context object. If called from Node, creates
 * a new object every time. Otherwise creates and caches an object in the browser and
 * subsequent calls will return that.
 *
 * @summary Gets a page context object
 * @returns {Object} Page context object
 */
export default function getPageContext() {
  // Make sure to create a new context for every server-side request so that data
  // isn't shared between connections (which would be bad).
  if (!process.browser) {
    return createPageContext();
  }

  // Reuse context on the client-side.
  if (!global.__INIT_MATERIAL_UI__) {
    global.__INIT_MATERIAL_UI__ = createPageContext();
  }

  return global.__INIT_MATERIAL_UI__;
}
