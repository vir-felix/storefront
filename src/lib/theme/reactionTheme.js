import { createMuiTheme } from "material-ui/styles";

const theme = createMuiTheme({
  palette: {
    primary: {
      light: "#26B0F9",
      main: "#1999DD",
      dark: "#172F3C",
      contrastText: "#FFFFFF",
      borderColor: "#CCCCCC"
    },
    secondary: {
      light: "#5d8ea9",
      main: "#5E7480",
      dark: "#1D1D1D",
      contrastText: "#000000"
    },
    error: {
      light: "#E54F5D",
      main: "#CD3F4C",
      dark: "#3C1F21",
      contrastText: "#FFFFFF"
    },
    reaction: {
      activeElementBorderColor: "#94E8D1",
      activeElementBackground: "#E6E6E6"
    }

  },
  typography: {
    fontFamily: "Source Sans Pro, Helvetica Neue, Helvetica, sans-serif",
    fontSize: 16,
    fontWeightLight: 200,
    fontWeightRegular: 400,
    fontWeightMedium: 700,
    fontWeightBold: 700
  }
});

export default theme;
