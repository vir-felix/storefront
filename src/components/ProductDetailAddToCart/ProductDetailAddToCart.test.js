import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "material-ui/styles";
import theme from "lib/theme/reactionTheme";
import ProductDetailAddToCart from "./ProductDetailAddToCart";


test("basic snapshot", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <ProductDetailAddToCart />
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
