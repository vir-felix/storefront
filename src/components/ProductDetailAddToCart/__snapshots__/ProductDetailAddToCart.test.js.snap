// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
Array [
  <div
    className="MuiGrid-container-8"
  >
    <div
      className="MuiGrid-item-9 MuiGrid-grid-xs-12-46 inject-ProductDetailAddToCart-with-uiStore-quantityGrid-5"
    >
      <div
        className="Divider-container-99"
      >
        <hr
          className="Divider-item-101"
        />
        <hr
          className="Divider-item-101"
        />
      </div>
      <span
        className="MuiTypography-root-102 MuiTypography-body1-111 inject-ProductDetailAddToCart-with-uiStore-quantityTypography-7"
      >
        Quantity
      </span>
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-126"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="MuiInput-root-130 inject-ProductDetailAddToCart-with-uiStore-quantityContainer-4 MuiInput-formControl-131"
        >
          <div
            className="MuiInputAdornment-root-143"
          >
            <button
              className="MuiButtonBase-root-146 inject-ProductDetailAddToCart-with-uiStore-incrementButton-3"
              color="default"
              disabled={undefined}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
              variant="outlined"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-149"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19,13H5V11H19V13Z"
                />
              </svg>
              <span
                className="MuiTouchRipple-root-155"
              />
            </button>
          </div>
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-138 inject-ProductDetailAddToCart-with-uiStore-quantityInput-6"
            defaultValue={undefined}
            disabled={false}
            id="addToCartQuantityInput"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value={1}
          />
          <div
            className="MuiInputAdornment-root-143"
          >
            <button
              className="MuiButtonBase-root-146 inject-ProductDetailAddToCart-with-uiStore-incrementButton-3"
              color="default"
              disabled={undefined}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
              variant="outlined"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-149"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                />
              </svg>
              <span
                className="MuiTouchRipple-root-155"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-9 MuiGrid-grid-xs-12-46"
    >
      <button
        className="MuiButtonBase-root-146 inject-ProductDetailAddToCart-with-uiStore-addToCartButton-1"
        disabled={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiTypography-root-102 MuiTypography-body1-111 inject-ProductDetailAddToCart-with-uiStore-addToCartText-2"
        >
          Add to cart
        </span>
        <span
          className="MuiTouchRipple-root-155"
        />
      </button>
    </div>
  </div>,
  <div
    className="HiddenCss-mdUp-177"
  >
    <div
      className="inject-CartPopover-with-uiStore-container-162 inject-CartPopover-with-uiStore-isContainerHidden-164"
    >
      <div
        className="MuiGrid-container-8 MuiGrid-spacing-xs-24-31 inject-CartPopover-with-uiStore-gridContainer-163"
      >
        <div
          className="MuiGrid-item-9 MuiGrid-grid-xs-12-46 inject-CartPopover-with-uiStore-containerItem-166"
        >
          <img
            alt="Item Title"
            className="inject-CartPopover-with-uiStore-addedToCartImg-167"
            src="http://placehold.it/100"
          />
          <span
            className="MuiTypography-root-102 MuiTypography-body1-111 inject-CartPopover-with-uiStore-addedToCartText-169"
          >
            10
             "
            <span
              className="inject-CartPopover-with-uiStore-addedToCartItemName-168"
            >
              Item Title
            </span>
            " added to cart
          </span>
        </div>
        <div
          className="MuiGrid-item-9 MuiGrid-grid-xs-12-46"
        >
          <a
            className="WithTracking-Link--link-185"
            href="/checkout"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="link"
            tabIndex={0}
            tracking={
              Object {
                "getTrackingData": [Function],
                "trackEvent": [Function],
              }
            }
          >
            <div
              className="Button__ButtonDiv-bnfrK eloVkm"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
              title={undefined}
            >
              <div>
                Checkout
              </div>
              <div
                className="Button__SpinnerWrap-bgOcuM cCKgPo"
                style={
                  Object {
                    "opacity": 0,
                    "paddingLeft": 0,
                    "width": 0,
                  }
                }
              >
                <svg
                  className="spinner__SpinnerSvg-iBTUvT bunzNs"
                  height="40px"
                  style={
                    Object {
                      "enableBackground": "new 0 0 50 50",
                    }
                  }
                  version="1.1"
                  viewBox="0 0 50 50"
                  width="40px"
                  x="0px"
                  y="0px"
                >
                  <path
                    d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
                    fill="#000"
                  >
                    <animateTransform
                      attributeName="transform"
                      attributeType="xml"
                      dur="1s"
                      from="0 25 25"
                      repeatCount="indefinite"
                      to="360 25 25"
                      type="rotate"
                    />
                  </path>
                </svg>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;
